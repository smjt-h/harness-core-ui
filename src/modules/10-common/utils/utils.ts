/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import type { IconName } from '@harness/uicore'
import { Scope } from '@common/interfaces/SecretsInterface'
import { ModuleName } from 'framework/types/ModuleName'

interface SetPageNumberProps {
  setPage: (value: React.SetStateAction<number>) => void
  pageItemsCount?: number
  page: number
}

export const getModuleIcon = (module: ModuleName): IconName => {
  switch (module) {
    case ModuleName.CD:
      return 'cd-main'
    case ModuleName.CV:
      return 'cv-main'
    case ModuleName.CI:
      return 'ci-main'
    case ModuleName.CE:
      return 'ce-main'
    case ModuleName.CF:
      return 'cf-main'
  }
  return 'nav-project'
}

export const getReference = (scope?: Scope, identifier?: string): string | undefined => {
  switch (scope) {
    case Scope.PROJECT:
      return identifier
    case Scope.ORG:
      return `org.${identifier}`
    case Scope.ACCOUNT:
      return `account.${identifier}`
  }
}

export const setPageNumber = ({ setPage, pageItemsCount, page }: SetPageNumberProps): void => {
  if (pageItemsCount === 0 && page > 0) {
    setPage(page - 1)
  }
}

export const formatCount = (num: number): string => {
  const min = 1e3
  if (num >= min) {
    const units = ['k', 'M', 'B', 'T']
    const order = Math.floor(Math.log(num) / Math.log(1000))
    const finalNum = Math.round(num / 1000 ** order)
    if (finalNum === min && order < units.length) {
      return 1 + units[order]
    }
    return finalNum + units[order - 1]
  }
  return num.toLocaleString()
}

export const isCommunityPlan = (): boolean => false

export const isOnPrem = (): boolean => window.deploymentType === 'ON_PREM'

export const focusOnNode = (node: HTMLElement): void => {
  const oldTabIndex = node.tabIndex
  node.tabIndex = -1
  node.focus()
  node.tabIndex = oldTabIndex
}

const HOTJAR_SUPPRESSION_ATTR = 'data-hj-suppress'

// Utility to add `data-hj-suppress` into a collection of elements to
// suppress data from HotJar recording
// @see https://bit.ly/3rCgpOY
export const suppressHotJarRecording = (elements: Element[] | null | undefined): void => {
  if (window.hj) {
    elements?.forEach?.((e: Element) => {
      if (!e.hasAttribute(HOTJAR_SUPPRESSION_ATTR)) {
        e.setAttribute(HOTJAR_SUPPRESSION_ATTR, 'true')
      }
    })
  }
}

// Utility to generate { 'data-hj-suppress': true } attribute if HotJar is available
export const addHotJarSuppressionAttribute = (): { [HOTJAR_SUPPRESSION_ATTR]: boolean } | undefined =>
  window.hj ? { [HOTJAR_SUPPRESSION_ATTR]: true } : undefined
