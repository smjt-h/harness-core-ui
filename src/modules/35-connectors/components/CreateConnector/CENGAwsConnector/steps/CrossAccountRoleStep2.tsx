/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React, { useEffect, useState } from 'react'
import { useParams } from 'react-router-dom'
import * as Yup from 'yup'
import {
  Button,
  Container,
  Formik,
  FormikForm,
  FormInput,
  Layout,
  ModalErrorHandler,
  ModalErrorHandlerBinding,
  getErrorInfoFromErrorObject,
  StepProps,
  Icon,
  Text,
  ButtonSize,
  ButtonVariation
} from '@wings-software/uicore'
import { FontVariation, Color } from '@harness/design-system'
import { pick } from 'lodash-es'
import { useStrings } from 'framework/strings'
import { CrossAccountAccess, useCreateConnector, useUpdateConnector, Failure } from 'services/cd-ng'
import { useAwsaccountconnectiondetail } from 'services/ce/index'
import { CE_AWS_CONNECTOR_CREATION_EVENTS } from '@connectors/trackingConstants'
import { useStepLoadTelemetry } from '@connectors/common/useTrackStepLoad/useStepLoadTelemetry'
import ConnectorInstructionList from '@connectors/common/ConnectorCreationInstructionList/ConnectorCreationInstructionList'
import { connectorHelperUrls } from '@connectors/constants'
import type { FeaturesString } from './CrossAccountRoleStep1'
import type { CEAwsConnectorDTO } from './OverviewStep'
import css from '../CreateCeAwsConnector.module.scss'

const CrossAccountRoleStep2: React.FC<StepProps<CEAwsConnectorDTO>> = props => {
  const { getString } = useStrings()

  useStepLoadTelemetry(CE_AWS_CONNECTOR_CREATION_EVENTS.LOAD_CREATE_CROSS_ACCOUNT_ROLE)

  const { accountId } = useParams<{
    accountId: string
  }>()
  const { prevStepData, nextStep, previousStep } = props
  const [externalId, setExternalId] = useState<string>('')
  const { mutate: createConnector } = useCreateConnector({
    queryParams: { accountIdentifier: accountId }
  })
  const { mutate: updateConnector } = useUpdateConnector({
    queryParams: { accountIdentifier: accountId }
  })
  const { data: awsUrlTemplateData, loading: awsUrlTemplateLoading } = useAwsaccountconnectiondetail({
    queryParams: { accountIdentifier: accountId }
  })
  const [modalErrorHandler, setModalErrorHandler] = useState<ModalErrorHandlerBinding | undefined>()
  const [isSubmitLoading, setIsSubmitLoading] = useState<boolean>(false)

  const randomString = (): string => {
    return Math.random().toString(36).substring(2, 8) + Math.random().toString(36).substring(2, 8)
  }

  useEffect(() => {
    if (awsUrlTemplateData?.status == 'SUCCESS' && !prevStepData?.isEditMode)
      setExternalId(awsUrlTemplateData?.data?.externalId || '')
  }, [awsUrlTemplateLoading])

  const featuresEnabled: FeaturesString[] = prevStepData?.spec?.featuresEnabled || []
  const curStatus = featuresEnabled.includes('BILLING')
  const visibiltyStatus = featuresEnabled.includes('VISIBILITY')
  const optimizationStatus = featuresEnabled.includes('OPTIMIZATION')

  const getRoleName = (roleArn: string) => {
    if (roleArn == undefined) return
    return roleArn.split('/')[1]
  }

  const makeTemplateUrl = () => {
    const baseurl = awsUrlTemplateData?.data?.stackLaunchTemplateLink
    const bucketName = prevStepData?.spec?.curAttributes?.s3BucketName || ''
    const rand = randomString()
    const roleName = prevStepData?.isEditMode
      ? getRoleName(prevStepData?.spec.crossAccountAccess.crossAccountRoleArn)
      : `HarnessCERole-${rand}`
    const url = `${baseurl}&param_ExternalId=${externalId}&param_BucketName=${bucketName}&param_BillingEnabled=${curStatus}&param_EventsEnabled=${visibiltyStatus}&param_OptimizationEnabled=${optimizationStatus}&param_RoleName=${roleName}`
    window.open(url)
  }

  const handleSubmit = async (formData: CrossAccountAccess) => {
    setIsSubmitLoading(true)
    const newspec = { ...prevStepData?.spec, crossAccountAccess: formData }
    if (prevStepData) {
      prevStepData.spec = newspec
    }

    try {
      if (prevStepData) {
        const connectorInfo: CEAwsConnectorDTO = {
          ...pick(prevStepData, ['name', 'identifier', 'description', 'tags', 'spec', 'type'])
        }
        if (prevStepData.isEditMode) {
          const response = await updateConnector({ connector: connectorInfo })
          if (response.status != 'SUCCESS') {
            throw response as Failure
          }
        } else {
          const response = await createConnector({ connector: connectorInfo })
          if (response.status != 'SUCCESS') {
            throw response as Failure
          }
        }
        nextStep?.(prevStepData)
      }
    } catch (e) {
      modalErrorHandler?.showDanger(getErrorInfoFromErrorObject(e))
    }
    setIsSubmitLoading(false)
  }

  const handleprev = () => {
    previousStep?.({ ...(prevStepData as CEAwsConnectorDTO) })
  }

  const instructionsList = [
    {
      type: 'hybrid',
      renderer: function instructionRenderer() {
        return (
          <>
            <Text font={{ variation: FontVariation.BODY }} color={Color.GREY_800}>
              {getString('connectors.ceAws.crossAccountRoleStep2.instructions.i1')}
              <Button
                text={getString('connectors.ceAws.crossAccountRoleStep2.launchTemplate')}
                rightIcon="main-share"
                variation={ButtonVariation.SECONDARY}
                size={ButtonSize.SMALL}
                iconProps={{ size: 12, margin: { left: 'xsmall' } }}
                onClick={() => {
                  makeTemplateUrl()
                }}
              />
            </Text>
            <Text font={{ variation: FontVariation.BODY }} color={Color.GREY_800}>
              {getString('connectors.ceAws.crossAccountRoleStep2.instructions.i2')}
              <a href={awsUrlTemplateData?.data?.cloudFormationTemplateLink || ''} target="_blank" rel="noreferrer">
                {getString('connectors.ceAws.crossAccountRoleStep2.instructions.templateRedirection')}
              </a>
              {'.)'}
            </Text>
          </>
        )
      }
    },
    {
      type: 'text',
      text: 'connectors.ceAws.crossAccountRoleStep2.instructions.i3',
      listClassName: 'instructionList'
    },
    {
      type: 'hybrid',
      listClassName: 'instructionList',
      renderer: function instructionRenderer() {
        return (
          <Text font={{ variation: FontVariation.BODY }} color={Color.GREY_800}>
            {getString('connectors.ceAws.cur.instructions.i2')}
            <a href={connectorHelperUrls.ceAwsRoleARNsteps} target="_blank" rel="noreferrer">
              {getString('connectors.ceAws.crossAccountRoleStep2.instructions.i4')}
              <Icon name="main-share" size={16} color={Color.PRIMARY_7} />
            </a>
          </Text>
        )
      }
    },
    {
      type: 'text',
      text: 'connectors.ceAws.crossAccountRoleStep2.instructions.i5',
      listClassName: 'instructionList'
    }
  ]

  return (
    <Layout.Vertical className={css.stepContainer}>
      <Text
        font={{ variation: FontVariation.H3 }}
        tooltipProps={{ dataTooltipId: 'awsConnectorCreateRole' }}
        margin={{ bottom: 'large' }}
      >
        {getString('connectors.ceAws.crossAccountRoleStep2.heading')}
      </Text>
      <Text color={Color.GREY_800} font={{ variation: FontVariation.BODY }} margin={{ bottom: 'large' }}>
        {getString('connectors.ceAws.crossAccountRoleStep2.subHeading')}
      </Text>
      <ConnectorInstructionList instructionsList={instructionsList} />
      <div style={{ flex: 1 }}>
        <Formik<CrossAccountAccess>
          formName="crossAccountRoleStep2Form"
          initialValues={{
            crossAccountRoleArn: prevStepData?.spec.crossAccountAccess.crossAccountRoleArn || '',
            externalId: prevStepData?.spec.crossAccountAccess.externalId || externalId || ''
          }}
          validationSchema={Yup.object().shape({
            crossAccountRoleArn: Yup.string()
              .matches(
                /arn:aws:iam::[0-9]*:role\/[^$%!&*?><\s]*/,
                getString('connectors.ceAws.crossAccountRoleStep2.validation.roleArnPattern')
              )
              .required(getString('connectors.ceAws.crossAccountRoleStep2.validation.roleArnRequired'))
          })}
          onSubmit={formData => {
            handleSubmit(formData)
          }}
          enableReinitialize={true}
        >
          {() => (
            <FormikForm>
              <ModalErrorHandler bind={setModalErrorHandler} />
              <Container className={css.main}>
                <FormInput.Text
                  name="crossAccountRoleArn"
                  label={getString('connectors.ceAws.crossAccountRoleStep2.roleArn')}
                  className={css.dataFields}
                  tooltipProps={{ dataTooltipId: 'crossAccountRoleArn' }}
                />

                <FormInput.Text
                  name="externalId"
                  label={getString('connectors.ceAws.crossAccountRoleStep2.extId')}
                  onChange={(e: React.ChangeEvent<HTMLInputElement>) => {
                    setExternalId(e.target.value)
                  }}
                  disabled={awsUrlTemplateLoading}
                  className={css.dataFields}
                  tooltipProps={{ dataTooltipId: 'externalId' }}
                />
              </Container>
              <a
                href={connectorHelperUrls.ceAwsNoAccount}
                target="_blank"
                rel="noreferrer"
                className={css.noAccountLink}
              >
                {getString('connectors.ceAws.crossAccountRoleStep2.dontHaveAccess')}
              </a>
              <Layout.Horizontal className={css.buttonPanel} spacing="small">
                <Button
                  text={getString('previous')}
                  icon="chevron-left"
                  onClick={handleprev}
                  disabled={isSubmitLoading}
                ></Button>
                <Button
                  type="submit"
                  intent="primary"
                  text={getString('saveAndContinue')}
                  rightIcon="chevron-right"
                  disabled={isSubmitLoading}
                />
                {isSubmitLoading && <Icon name="spinner" size={24} color="blue500" />}
              </Layout.Horizontal>
            </FormikForm>
          )}
        </Formik>
      </div>
    </Layout.Vertical>
  )
}

export default CrossAccountRoleStep2
