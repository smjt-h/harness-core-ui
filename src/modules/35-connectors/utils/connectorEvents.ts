/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import { Connectors } from '@connectors/constants'
import {
  CE_AWS_CONNECTOR_CREATION_EVENTS,
  CE_AZURE_CONNECTOR_CREATION_EVENTS,
  CE_K8S_CONNECTOR_CREATION_EVENTS
} from '@connectors/trackingConstants'

export const connectorsTrackEventMap = {
  [Connectors.CE_KUBERNETES]: CE_K8S_CONNECTOR_CREATION_EVENTS.LOAD_CONNECTION_TEST,
  [Connectors.CEAWS]: CE_AWS_CONNECTOR_CREATION_EVENTS.LOAD_CONNECTION_TEST,
  [Connectors.CE_AZURE]: CE_AZURE_CONNECTOR_CREATION_EVENTS.LOAD_CONNECTION_TEST
}
