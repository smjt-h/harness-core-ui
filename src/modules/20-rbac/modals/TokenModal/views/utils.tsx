/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import { getReadableDateTime } from '@common/utils/dateUtils'

export const DATE_FORMAT = 'MM/DD/YYYY'

export const hasNoExpiryDate = (value?: number): boolean => {
  return getReadableDateTime(value, DATE_FORMAT) === '12/31/2099'
}

export const getSelectedExpiration = (value?: number): string => {
  if (!value) {
    return '30'
  } else if (hasNoExpiryDate(value)) {
    return '-1'
  }
  return 'custom'
}

export const getSelectedExpirationDate = (value: string): string => {
  switch (value) {
    case '30':
    case '90':
    case '180':
      return getReadableDateTime(new Date().setDate(new Date().getDate() + parseInt(value)), DATE_FORMAT)
    case '-1':
      return '12/31/2099'
    default:
      return ''
  }
}

export const getExpirationDate = (value?: number): string => {
  if (value) {
    return getReadableDateTime(value, DATE_FORMAT)
  } else {
    return getSelectedExpirationDate(getSelectedExpiration(value))
  }
}
