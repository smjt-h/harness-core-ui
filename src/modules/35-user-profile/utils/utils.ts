/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import type { IconName } from '@wings-software/uicore'
import { ValueType } from '@secrets/components/TextReference/TextReference'
import type { SCMData } from '@user-profile/modals/SourceCodeManager/views/SourceCodeManagerForm'
import type { SourceCodeManagerAuthentication, SourceCodeManagerDTO } from 'services/cd-ng'
import { setSecretField } from '@secrets/utils/SecretField'

export enum ConnectionType {
  HTTPS = 'HTTPS',
  HTTP = 'Http',
  SSH = 'Ssh'
}

export enum AuthTypes {
  USERNAME_PASSWORD = 'UsernamePassword',
  USERNAME_TOKEN = 'UsernameToken',
  KERBEROS = 'Kerberos',
  SSH_KEY = 'SSH_KEY',
  AWSCredentials = 'AWSCredentials'
}

export interface SourceCodeType {
  text: string
  value: SourceCodeTypes
  icon: IconName
}

export enum SourceCodeTypes {
  BITBUCKET = 'BITBUCKET',
  GITHUB = 'GITHUB',
  GITLAB = 'GITLAB',
  AZURE_REPO = 'AZURE_REPO',
  AWS_CODE_COMMIT = 'AWS_CODE_COMMIT'
}

export const selectedValueToTypeMap: Record<SourceCodeTypes, SourceCodeTypes> = {
  [SourceCodeTypes.BITBUCKET]: SourceCodeTypes.BITBUCKET,
  [SourceCodeTypes.GITHUB]: SourceCodeTypes.GITHUB,
  [SourceCodeTypes.GITLAB]: SourceCodeTypes.GITLAB,
  [SourceCodeTypes.AWS_CODE_COMMIT]: SourceCodeTypes.AWS_CODE_COMMIT,
  [SourceCodeTypes.AZURE_REPO]: SourceCodeTypes.AZURE_REPO
}

export const getIconBySCM = (item: SourceCodeTypes): IconName => {
  switch (item) {
    case SourceCodeTypes.BITBUCKET:
      return 'bitbucket-blue'
    case SourceCodeTypes.GITHUB:
      return 'github'
    case SourceCodeTypes.GITLAB:
      return 'service-gotlab'
    case SourceCodeTypes.AWS_CODE_COMMIT:
      return 'service-aws-code-deploy'
    case SourceCodeTypes.AZURE_REPO:
      return 'service-azure'
    default:
      return 'bitbucket'
  }
}

export const getAuthentication = (values: SCMData): SourceCodeManagerAuthentication | undefined => {
  switch (values.authType) {
    case AuthTypes.USERNAME_PASSWORD:
      return {
        type: ConnectionType.HTTP,
        spec: {
          type: AuthTypes.USERNAME_PASSWORD,
          spec: {
            ...(values.username?.type === ValueType.TEXT
              ? { username: values.username.value }
              : { usernameRef: values.username?.value }),
            passwordRef: values.password?.referenceString
          }
        }
      }
    case AuthTypes.USERNAME_TOKEN:
      return {
        type: ConnectionType.HTTP,
        spec: {
          type: AuthTypes.USERNAME_TOKEN,
          spec: {
            ...(values.username?.type === ValueType.TEXT
              ? { username: values.username.value }
              : { usernameRef: values.username?.value }),
            tokenRef: values.accessToken?.referenceString
          }
        }
      }
    default:
      return undefined
  }
}

const getBitBucketFormData = async (
  sourceCodeManagerData: SourceCodeManagerDTO,
  accountIdentifier: string
): Promise<SCMData> => {
  const { name, authentication } = sourceCodeManagerData
  return {
    name: name,
    authType: authentication?.spec?.type,
    password: (await setSecretField(authentication?.spec?.spec?.passwordRef, {
      accountIdentifier
    })) as SCMData['password'],
    username: {
      value: authentication?.spec?.spec?.username || authentication?.spec?.spec?.usernameRef,
      type: authentication?.spec?.spec?.usernameRef ? ValueType.ENCRYPTED : ValueType.TEXT
    }
  }
}

const getGithubFormData = async (
  sourceCodeManagerData: SourceCodeManagerDTO,
  accountIdentifier: string
): Promise<SCMData> => {
  const { name, authentication } = sourceCodeManagerData
  return {
    name: name,
    authType: authentication?.spec?.type,
    accessToken: (await setSecretField(
      authentication?.spec?.spec?.tokenRef || authentication?.spec?.apiAccess?.spec?.tokenRef,
      { accountIdentifier }
    )) as SCMData['accessToken'],
    username: {
      value: authentication?.spec?.spec?.username || authentication?.spec?.spec?.usernameRef,
      type: authentication?.spec?.spec?.usernameRef ? ValueType.ENCRYPTED : ValueType.TEXT
    }
  }
}

export const getFormDataBasedOnSCMType = (
  sourceCodeManagerData: SourceCodeManagerDTO,
  accountIdentifier: string
): Promise<SCMData | undefined> => {
  switch (sourceCodeManagerData.type) {
    case SourceCodeTypes.BITBUCKET:
      return getBitBucketFormData(sourceCodeManagerData, accountIdentifier)
    case SourceCodeTypes.GITHUB:
      return getGithubFormData(sourceCodeManagerData, accountIdentifier)
    default:
      return Promise.resolve(undefined)
  }
}

export const getDefaultSCMType = (
  supportedSCMS: SourceCodeType[],
  currentSCM?: SourceCodeManagerDTO['type']
): SourceCodeType => {
  return supportedSCMS.find(item => item.value === currentSCM) || supportedSCMS[0]
}

export const getDefaultSelected = (type?: SourceCodeTypes): AuthTypes | undefined => {
  switch (type) {
    case SourceCodeTypes.GITHUB:
      return AuthTypes.USERNAME_TOKEN
    case SourceCodeTypes.BITBUCKET:
    case SourceCodeTypes.GITLAB:
    case SourceCodeTypes.AZURE_REPO:
      return AuthTypes.USERNAME_PASSWORD
    case SourceCodeTypes.AWS_CODE_COMMIT:
      return AuthTypes.AWSCredentials
    default:
      return undefined
  }
}
