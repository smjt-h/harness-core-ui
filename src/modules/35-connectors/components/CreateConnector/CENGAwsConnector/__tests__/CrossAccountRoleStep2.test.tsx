/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { act, fireEvent, render } from '@testing-library/react'
import { renderHook } from '@testing-library/react-hooks'
import { TestWrapper } from '@common/utils/testUtils'

import { useStrings } from 'framework/strings'
import CrossAccountRoleStep2 from '../steps/CrossAccountRoleStep2'

jest.mock('services/ce/index', () => ({
  useAwsaccountconnectiondetail: jest.fn().mockImplementation(() => ({
    mutate: async () => {
      return {
        status: 'SUCCESS'
      }
    },
    loading: false
  }))
}))

const wrapper = ({ children }: React.PropsWithChildren<unknown>): React.ReactElement => (
  <TestWrapper>{children}</TestWrapper>
)

const { result } = renderHook(() => useStrings(), { wrapper })

describe('Create CE AWS Connector Create Cross Account Role Step', () => {
  test('should render form', async () => {
    window.open = jest.fn()

    const { container, getByText } = render(
      <TestWrapper>
        <CrossAccountRoleStep2 name={result.current.getString('connectors.ceAws.steps.roleARN')} />
      </TestWrapper>
    )

    expect(container).toMatchSnapshot()

    act(() => {
      fireEvent.click(getByText('connectors.ceAws.crossAccountRoleStep2.launchTemplate'))
    })

    act(() => {
      fireEvent.click(getByText('previous'))
    })
  })
})
