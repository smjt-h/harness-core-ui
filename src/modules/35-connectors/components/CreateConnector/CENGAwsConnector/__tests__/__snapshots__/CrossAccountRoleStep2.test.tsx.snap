// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create CE AWS Connector Create Cross Account Role Step should render form 1`] = `
<div>
  <div
    class="Layout--vertical StyledProps--main stepContainer"
  >
    <div
      class="Text--withDocsTooltip"
      data-tooltip-id="awsConnectorCreateRole"
    >
      <p
        class="StyledProps--font StyledProps--main StyledProps--font-variation-h3 StyledProps--margin-bottom-large"
      >
        connectors.ceAws.crossAccountRoleStep2.heading
      </p>
    </div>
    <p
      class="StyledProps--font StyledProps--main StyledProps--color StyledProps--color-grey800 StyledProps--font-variation-body StyledProps--margin-bottom-large"
    >
      connectors.ceAws.crossAccountRoleStep2.subHeading
    </p>
    <div
      class="StyledProps--font StyledProps--main instructionPanel"
    >
      <ol
        class="instructionList"
      >
        <li
          class="undefined"
        >
          <p
            class="StyledProps--font StyledProps--main StyledProps--font-variation-body StyledProps--color StyledProps--color-grey800"
          >
            connectors.ceAws.crossAccountRoleStep2.instructions.i1
            <button
              class="bp3-button Button--button StyledProps--font StyledProps--main Button--with-current-color Button--withRightIcon Button--variation Button--variation-secondary Button--size Button--small"
              type="button"
            >
              <span
                class="bp3-button-text"
              >
                connectors.ceAws.crossAccountRoleStep2.launchTemplate
              </span>
              <span
                class="bp3-icon StyledProps--main StyledProps--margin-left-xsmall"
                data-icon="main-share"
              >
                <svg
                  height="12"
                  viewBox="0 0 14 14"
                  width="12"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M10.5 12.6H1.4V3.5h6.1l1.4-1.4H.7c-.4 0-.7.3-.7.7v10.5c0 .4.3.7.7.7h10.5c.4 0 .7-.3.7-.7V5.1l-1.4 1.4zM13.3 0H8.4c-.4 0-.7.3-.7.7s.3.7.7.7h3.2L6.5 6.5c-.1.1-.2.3-.2.5 0 .4.3.7.7.7.2 0 .4-.1.5-.2l5.1-5.1v3.2c0 .4.3.7.7.7s.7-.3.7-.7V.7c0-.4-.3-.7-.7-.7z"
                    fill="#4d5969"
                  />
                </svg>
              </span>
            </button>
          </p>
          <p
            class="StyledProps--font StyledProps--main StyledProps--font-variation-body StyledProps--color StyledProps--color-grey800"
          >
            connectors.ceAws.crossAccountRoleStep2.instructions.i2
            <a
              href=""
              rel="noreferrer"
              target="_blank"
            >
              connectors.ceAws.crossAccountRoleStep2.instructions.templateRedirection
            </a>
            .)
          </p>
        </li>
        <li
          class="instructionList"
        >
          <p
            class="StyledProps--font StyledProps--main StyledProps--font-variation-body StyledProps--color StyledProps--color-grey800"
          >
            connectors.ceAws.crossAccountRoleStep2.instructions.i3
          </p>
        </li>
        <li
          class="instructionList"
        >
          <p
            class="StyledProps--font StyledProps--main StyledProps--font-variation-body StyledProps--color StyledProps--color-grey800"
          >
            connectors.ceAws.cur.instructions.i2
            <a
              href="https://ngdocs.harness.io/article/80vbt5jv0q-set-up-cost-visibility-for-aws#step_4_create_cross_account_role"
              rel="noreferrer"
              target="_blank"
            >
              connectors.ceAws.crossAccountRoleStep2.instructions.i4
              <span
                class="bp3-icon StyledProps--main StyledProps--color StyledProps--color-primary7"
                data-icon="main-share"
              >
                <svg
                  height="16"
                  viewBox="0 0 14 14"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M10.5 12.6H1.4V3.5h6.1l1.4-1.4H.7c-.4 0-.7.3-.7.7v10.5c0 .4.3.7.7.7h10.5c.4 0 .7-.3.7-.7V5.1l-1.4 1.4zM13.3 0H8.4c-.4 0-.7.3-.7.7s.3.7.7.7h3.2L6.5 6.5c-.1.1-.2.3-.2.5 0 .4.3.7.7.7.2 0 .4-.1.5-.2l5.1-5.1v3.2c0 .4.3.7.7.7s.7-.3.7-.7V.7c0-.4-.3-.7-.7-.7z"
                    fill="#4d5969"
                  />
                </svg>
              </span>
            </a>
          </p>
        </li>
        <li
          class="instructionList"
        >
          <p
            class="StyledProps--font StyledProps--main StyledProps--font-variation-body StyledProps--color StyledProps--color-grey800"
          >
            connectors.ceAws.crossAccountRoleStep2.instructions.i5
          </p>
        </li>
      </ol>
    </div>
    <div
      style="flex: 1;"
    >
      <div
        class="OverlaySpinner--overlaySpinner"
      >
        <form
          action="#"
          class="FormikForm--main"
        >
          <div>
            <div
              class="StyledProps--font StyledProps--main main"
            >
              <div
                class="bp3-form-group dataFields"
                data-id="crossAccountRoleArn-0"
              >
                <label
                  class="bp3-label"
                  for="crossAccountRoleArn"
                >
                  <span
                    class="Tooltip--acenter"
                    data-tooltip-id="crossAccountRoleArn"
                  >
                    connectors.ceAws.crossAccountRoleStep2.roleArn
                  </span>
                   
                  <span
                    class="bp3-text-muted"
                  />
                </label>
                <div
                  class="bp3-form-content"
                >
                  <div
                    class="bp3-input-group"
                  >
                    <input
                      autocomplete="off"
                      class="bp3-input"
                      name="crossAccountRoleArn"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </div>
              <div
                class="bp3-form-group dataFields"
                data-id="externalId-1"
              >
                <label
                  class="bp3-label"
                  for="externalId"
                >
                  <span
                    class="Tooltip--acenter"
                    data-tooltip-id="externalId"
                  >
                    connectors.ceAws.crossAccountRoleStep2.extId
                  </span>
                   
                  <span
                    class="bp3-text-muted"
                  />
                </label>
                <div
                  class="bp3-form-content"
                >
                  <div
                    class="bp3-input-group"
                  >
                    <input
                      autocomplete="off"
                      class="bp3-input"
                      name="externalId"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </div>
            </div>
            <a
              class="noAccountLink"
              href="https://newdocs.helpdocs.io/article/80vbt5jv0q-set-up-cost-visibility-for-aws#review_aws_access_permissions"
              rel="noreferrer"
              target="_blank"
            >
              connectors.ceAws.crossAccountRoleStep2.dontHaveAccess
            </a>
            <div
              class="Layout--horizontal Layout--layout-spacing-small StyledProps--main buttonPanel"
            >
              <button
                class="bp3-button Button--button StyledProps--font StyledProps--main Button--with-current-color Button--withLeftIcon"
                type="button"
              >
                <span
                  class="bp3-icon bp3-icon-chevron-left StyledProps--main StyledProps--padding-right-xsmall"
                  icon="chevron-left"
                >
                  <svg
                    data-icon="chevron-left"
                    height="16"
                    viewBox="0 0 16 16"
                    width="16"
                  >
                    <desc>
                      chevron-left
                    </desc>
                    <path
                      d="M7.41 8l3.29-3.29c.19-.18.3-.43.3-.71a1.003 1.003 0 00-1.71-.71l-4 4C5.11 7.47 5 7.72 5 8c0 .28.11.53.29.71l4 4a1.003 1.003 0 001.42-1.42L7.41 8z"
                      fill-rule="evenodd"
                    />
                  </svg>
                </span>
                <span
                  class="bp3-button-text"
                >
                  previous
                </span>
              </button>
              <button
                class="bp3-button Button--button StyledProps--font StyledProps--main StyledProps--intent-primary Button--with-current-color Button--withRightIcon"
                type="submit"
              >
                <span
                  class="bp3-button-text"
                >
                  saveAndContinue
                </span>
                <span
                  class="bp3-icon bp3-icon-chevron-right StyledProps--main"
                  icon="chevron-right"
                >
                  <svg
                    data-icon="chevron-right"
                    height="16"
                    viewBox="0 0 16 16"
                    width="16"
                  >
                    <desc>
                      chevron-right
                    </desc>
                    <path
                      d="M10.71 7.29l-4-4a1.003 1.003 0 00-1.42 1.42L8.59 8 5.3 11.29c-.19.18-.3.43-.3.71a1.003 1.003 0 001.71.71l4-4c.18-.18.29-.43.29-.71 0-.28-.11-.53-.29-.71z"
                      fill-rule="evenodd"
                    />
                  </svg>
                </span>
              </button>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
